src/testRunner/unittests/tsserver/smartSelection.ts

```diff
@@ -0,0 +1,687 @@
+namespace ts.projectSystem {
+    function setup(fileName: string, content: string) {
+        const file: File = { path: fileName, content };
+        const host = createServerHost([file, libFile]);
+        const session = createSession(host);
+        openFilesForSession([file], session);
+        return function getSmartSelectionRange(locations: protocol.SelectionRangeRequestArgs["locations"]) {
+            return executeSessionRequest<protocol.SelectionRangeRequest, protocol.SelectionRangeResponse>(
+                session,
+                CommandNames.SelectionRange,
+                { file: fileName, locations });
+        };
+    }
+
+    describe("unittests:: tsserver:: smartSelection", () => {
+        it("works for simple JavaScript", () => {
+            const getSmartSelectionRange = setup("/file.js", `

```

Comment:
So having couple of tests here to verify tsserver API is good idea but rest of them should be fourslash test as its mainly language service functionality you are testing.

Reply:
What exactly is the motivation for moving them to fourslash? Having them here, with the code in template strings, has the advantage that you can read line and offset from your editor. In fourslash, offset would be difficult to use because of the four slashes and marker characters. As an alternative, I wrote a utility to convert these existing tests to fourslash ranges, thinking that might be easier to write and read, but I think it's worse, at least to read:

```
[|type M = [|{ [|[|-readonly [K in keyof any][|-[|/**/?|]|]|]: any|] }|];|]|]
```

I get that fourslash is one step closer to the language service, but I'm not sure it's a good format for assertions about this. Have any suggestions?

Reply:
@sheetalkamat done. Created a baseline format for these.